src/components/main/CopyButton.tsx

```diff
@@ -1,56 +1,114 @@
-import { MouseEventHandler } from 'react';
+import { useEffect, useState } from 'react';
 
 import { IconButtonProps } from '@mui/material';
 
+import { DiscriminatedItem } from '@graasp/sdk';
 import {
   ActionButtonVariant,
   CopyButton as GraaspCopyButton,
 } from '@graasp/ui';
 
+import { mutations } from '@/config/queryClient';
+import { computeButtonText } from '@/utils/itemSelection';
+
 import { useBuilderTranslation } from '../../config/i18n';
 import {
   ITEM_COPY_BUTTON_CLASS,
   ITEM_MENU_COPY_BUTTON_CLASS,
 } from '../../config/selectors';
 import { BUILDER } from '../../langs/constants';
-import { useCopyItemModalContext } from '../context/CopyItemModalContext';
+import { NavigationElement } from './itemSelectionModal/Breadcrumbs';
+import ItemSelectionModal, {
+  ItemSelectionModalProps,
+} from './itemSelectionModal/ItemSelectionModal';
 
 export type Props = {
   color?: IconButtonProps['color'];
   id?: string;
-  onClick?: MouseEventHandler<HTMLButtonElement | HTMLLIElement>;
+  onClick?: () => void;
   type?: ActionButtonVariant;
   itemIds: string[];
 };
 
 const CopyButton = ({
-  itemIds,
+  itemIds: defaultItemsIds,
```

Comment:
Not sure why you receive some itemIds here ?

Reply:
I think, this is the same reason as for the MoveButton. I just based my code on this one.
The given reason in comments is `necessary to sync prop with a state because move-many-items' targets are updated dynamically with the table`.